pkg/webhook/machineconfigpool_webhook.go

```diff
@@ -0,0 +1,139 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/klog/v2"
+	"k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	machinecfgv1 "github.com/openshift/api/machineconfiguration/v1"
+	mcfglistersv1alpha1 "github.com/openshift/client-go/machineconfiguration/listers/machineconfiguration/v1alpha1"
+)
+
+var _ admission.CustomValidator = (*machineConfigPoolValidatorHandler)(nil)
+
+type admissionConfig struct {
```

Comment:
typically, I find you will probably end up needing these types outside of the webhook pkg lol. No harm in exporting them